backend/src/test/java/com/google/models/AlertTest.java

```diff
@@ -0,0 +1,112 @@
+import org.junit.runners.JUnit4;
+
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import java.util.*;
+import com.google.blackswan.mock.*;
+
+/** Contain tests for methods in {@link Alert} class. */
+@RunWith(JUnit4.class)
+public final class AlertTest {
+  private static final int TIMESTAMP_CONSTANT = 1;
+  private static final AnomalyGenerator dummyAnomalyGenerator = new DummyAnomalyGenerator();
+  private static final LocalServiceTestHelper helper = 
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  private Alert alert;
+
+  @Before
+  public void setUp() throws Exception {
+    helper.setUp();
+    // TODO: Find out whether it's better to set parameters for Alert as private static variables,
+    //       since they're used later to test getters. 
+    alert = new Alert(Timestamp.getDummyTimestamp(TIMESTAMP_CONSTANT), dummyAnomalyGenerator.getAnomalies(), 
+        Alert.UNRESOLVED_MESSAGE);
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void getTimestamp_workingGetter() {
+    assertEquals(alert.getTimestamp(), Timestamp.getDummyTimestamp(TIMESTAMP_CONSTANT));
+  }
+
+  @Test
+  public void getAnomalies_workingGetter() {
+    // Check to see if List<Anomaly> returned is same as List<Anomaly> used to generate alert.
+    assertEquals(alert.getAnomalies(), dummyAnomalyGenerator.getAnomalies());
+  }
+
+  @Test
+  public void getResolvedStatus_workingGetter() {
+    assertEquals(alert.getResolvedStatus(), Alert.UNRESOLVED_MESSAGE);
+  }
+
+  @Test
+  public void setResolvedStatus_workingSetter() {
+    alert.setResolvedStatus(Alert.RESOLVED_MESSAGE);
+
+    assertEquals(alert.getResolvedStatus(), Alert.RESOLVED_MESSAGE);
+  }
+
+  @Test
+  public void equals_workingComparator() {
+    Alert sameAlert = new Alert(Timestamp.getDummyTimestamp(TIMESTAMP_CONSTANT), dummyAnomalyGenerator.getAnomalies(), 
+        Alert.UNRESOLVED_MESSAGE);
+    Alert diffTimeAlert = new Alert(Timestamp.getDummyTimestamp(TIMESTAMP_CONSTANT + 1), dummyAnomalyGenerator.getAnomalies(), 
+        Alert.UNRESOLVED_MESSAGE);
+    Alert diffResolveAlert = new Alert(Timestamp.getDummyTimestamp(TIMESTAMP_CONSTANT), dummyAnomalyGenerator.getAnomalies(), 
+        Alert.RESOLVED_MESSAGE);
+    // TODO: Test equals with Alert object that has different list of anomalies. Currently, dummyAnomalyGenerator can only 
+    //       generate one list of anomalies right now. 
+
+    assertTrue(alert.equals(alert));
+    assertFalse(alert.equals(diffTimeAlert));
+    assertFalse(alert.equals(diffResolveAlert));
+    assertTrue(alert.equals(sameAlert));
```

Comment:
nit: move this test after the assertTrue one on line 81. was a little confusing to understand the tests